components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/ApplicationPurge.java

```diff
@@ -15,16 +15,115 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
 package org.wso2.carbon.apimgt.cleanup.service;
 
+import com.google.gson.Gson;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+
 import java.util.LinkedHashMap;
+import java.util.Map;
 
 public class ApplicationPurge implements OrganizationPurge {
-    public LinkedHashMap<String, String> deleteOrganization(String organization) {
-        return null;
+    private String username;
+    protected OrganizationPurgeDAO organizationPurgeDAO;
+    private static final Log log = LogFactory.getLog(ApplicationPurge.class);
+    LinkedHashMap<String, String> applicationPurgeTaskMap = new LinkedHashMap<>();
+
+    private void initTaskList() {
+        applicationPurgeTaskMap.put(APIConstants.OrganizationDeletion.PENDING_SUBSCRIPTION_REMOVAL,
+                APIConstants.OrganizationDeletion.PENDING);
+        applicationPurgeTaskMap.put(APIConstants.OrganizationDeletion.APPLICATION_CREATION_WF_REMOVAL,
+                APIConstants.OrganizationDeletion.PENDING);
+        applicationPurgeTaskMap.put(APIConstants.OrganizationDeletion.APPLICATION_REGISTRATION_REMOVAL,
+                APIConstants.OrganizationDeletion.PENDING);
+        applicationPurgeTaskMap.put(APIConstants.OrganizationDeletion.APPLICATION_REMOVAL,
+                APIConstants.OrganizationDeletion.PENDING);
+    }
+
+    public ApplicationPurge(String username) {
+        this.username = username;
+        organizationPurgeDAO = OrganizationPurgeDAO.getInstance();
+        initTaskList();
+    }
+
+    public ApplicationPurge(OrganizationPurgeDAO organizationPurgeDAO, String username) {
+        this.organizationPurgeDAO = organizationPurgeDAO;
```

Comment:
call as this and initialize upper constructor

Reply:
@rashm1n  Could you please fix this?

Reply:
@HiranyaKavishani this is already fixed. Resolving.